src/main/capi/stream-c.cpp

```diff
@@ -7,19 +7,28 @@ duckdb_data_chunk duckdb_stream_fetch_chunk(duckdb_result result) {
 		return nullptr;
 	}
 	auto &result_data = *((duckdb::DuckDBResultData *)result.internal_data);
-	if (result_data.result_set_type == duckdb::CAPIResultSetType::CAPI_RESULT_TYPE_DEPRECATED) {
```

Comment:
Does this mean we are entirely removing the old CAPI_RESULT_TYPE_DEPRECATED ?
There's more clean up to be done or possibly schedule for a future PR if that's the case